### PR TITLE
Qmaps 419  metadata

### PR DESCRIPTION
**Pull request by Julien-laville**
Add og:meta to home page (description, logo, title, locale, image, site_name)

### CODE REVIEW

File: bin/middlewares/og_meta.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+const request = require('request')
+const acceptLanguage = require('accept-language')
+const QueryError = require('../query_error')
+
+const metas = [
+  {name : 'site_name', content : 'Qwant map'},
```
Comment (amatissart) [inline bin/middlewares/og_meta.js]: "Qwant map" :arrow_right: "Qwant Maps"

File: bin/middlewares/og_meta.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  {name : 'image', content : '/images/maps_opengraph.png'},
+  {name : 'url', content : 'www.qwant.com/maps'},
```
Comment (amatissart) [inline bin/middlewares/og_meta.js]: Could the url be set dynamically based on the request ?

File: bin/middlewares/og_meta.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    res.locals.metas.push({name : 'title', content : `${res.locals. _('Qwant map')} | ${poi.name}`})
+    next()
+  }
+  function homeMeta(locale, req, res, next) {
+    commonMeta(locale, req, res)
+    res.locals.metas.push({name : 'title', content : `${res.locals. _('Qwant map')}`})
+    next()
+  }
+
+  function getLocale(req) {
```
Comment (amatissart) [inline bin/middlewares/og_meta.js]: This function is not specific to opengraph and is maybe redundant with the existing `user_language.js` middleware. 
It should be refactored there.

File: bin/middlewares/og_meta.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          resolve(null)
+        } else {
+          resolve(body)
+        }
+      })
+    })
+  }
+
+  function commonMeta(locale, req, res) {
+    res.locals.metas = metas.map(meta => meta)
```
Comment (amatissart) [inline bin/middlewares/og_meta.js]: maybe `ogMetas` instead of `metas` in locals ?

File: bin/middlewares/og_meta.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    res.locals.metas.push({name : 'locale', content : locale.locale})
+    res.locals.metas.push({name : 'description', content : res.locals. _('Map multiple locations. Do more with Qwant Maps.')})
+  }
+
+  function poiMeta(poi, locale, req, res, next) {
+    commonMeta(locale, req, res)
+    res.locals.poi = poi
+    res.locals.metas.push({name : 'title', content : `${res.locals. _('Qwant map')} | ${poi.name}`})
```
Comment (amatissart) [inline bin/middlewares/og_meta.js]: The poi name is probably enough in the title, as "Qwant Maps" is already defined as the site name

File: bin/middlewares/og_meta.js (hunk 1 above)
Comment (Julien-laville) [reply bin/middlewares/og_meta.js]: done

File: bin/middlewares/og_meta.js (hunk 2 above)
Comment (Julien-laville) [reply bin/middlewares/og_meta.js]: Je veux pas le faire car non trivial 
l'url se genere avec `${req.protocol}://${req.get('host')}` mais alors il manque la query (/maps)
mais si j'ajoute queryUrl alors on aura aussi /place à la suite

File: bin/middlewares/og_meta.js (hunk 3 above)
Comment (Julien-laville) [reply bin/middlewares/og_meta.js]: cleaned

File: bin/middlewares/og_meta.js (hunk 4 above)
Comment (Julien-laville) [reply bin/middlewares/og_meta.js]: refactored

File: bin/middlewares/og_meta.js (hunk 5 above)
Comment (Julien-laville) [reply bin/middlewares/og_meta.js]: ok but i keep qwant map as title if there is no poi

File: tests/integration/tests/poi.js [hunk 6]
```diff
@@ -32,6 +32,7 @@ test('click on a poi', async () => {
   await page.goto(APP_URL)
   await page.evaluate(() => {
     window.MAP_MOCK.evented.prepare('click', 'poi-level-1',  {originalEvent : {clientX : 1000},features : [{properties :{global_id : 1}}]})
+    window.MAP_MOCK.evented.prepare('click', 'poi-level-1',  {originalEvent : {clientX : 1000},features : [{properties :{global_id : 1}}]})
```
Comment (amatissart) [inline tests/integration/tests/poi.js]: Copy paste error ?

Comment (amatissart) [inline tests/integration/tests/poi.js]: Copy paste error ?

File: bin/middlewares/og_meta.js (hunk 2 above)
Comment (amatissart) [reply bin/middlewares/og_meta.js]: I am not sure. We also need to set a canonical url when the page is about a specific poi.

File: tests/integration/tests/poi.js (hunk 6 above)
Comment (Julien-laville) [reply tests/integration/tests/poi.js]: fixed with rebase

**Pull request merged**